magda-gateway/src/createHttpsRedirectionMiddleware.ts

```diff
@@ -0,0 +1,27 @@
+import { NextFunction, Response, Request, RequestHandler } from "express";
+import * as URI from "urijs";
+
+export default function createHttpsRedirectionMiddleware(
+    enableHttpsRedirection: boolean
+): RequestHandler {
+    return function(req: Request, res: Response, next: NextFunction) {
+        if (!enableHttpsRedirection) {
+            next();
+            return;
+        }
+
+        const uri = new URI(req.originalUrl);
+        if (uri.pathname() === "/v0/healthz") {
+            next();
+            return;
+        }
+
+        const protocol = req.get("X-Forwarded-Proto");
+        if (protocol && protocol === "http") {
```

Comment:
Does this mean that it won't redirect unless the connection is coming through some kind of proxy that sets `X-Forwarded-Proto`?

Is it possible that someone will want HTTPS redirection but not have the header? I guess it always has to go through the ingress in our current config?

In any case I think http://expressjs.com/en/api.html#req.protocol might be better?

Reply:
👍 to `req.protocol`. And set your [trust proxy](https://expressjs.com/en/guide/behind-proxies.html) appropriately so that `req.protocol` and other properties use trustworthy `X-Forwarded-*` headers.